src/com/ansorgit/plugins/bash/lang/lexer/bash.flex

```diff
@@ -168,27 +168,34 @@ Filedescriptor = "&" {IntegerLiteral} | "&-"
 %xstate X_HEREDOC_MARKER
 %xstate X_HEREDOC_MARKER_IGNORE_TABS
 %xstate X_HEREDOC
+// this state is used to keep the lexer in a non-zero state when heredoc markers were found
+// if used as a highlighting lexer incremental lexing is used when in a zero state
+// heredoc lexing is stateful and is thus not compatible with the zero state
+%state S_HEREDOC_EXPECTED
 
 /* To match here-strings */
 %xstate X_HERE_STRING
 
 %%
 /***************************** INITIAL STAATE ************************************/
-<YYINITIAL, S_CASE, S_CASE_PATTERN, S_SUBSHELL, S_ASSIGNMENT_LIST> {
+<YYINITIAL, S_CASE, S_CASE_PATTERN, S_SUBSHELL, S_ASSIGNMENT_LIST, S_HEREDOC_EXPECTED> {
   {Shebang}                     { return SHEBANG; }
   {Comment}                     { return COMMENT; }
 }
 
 <X_HEREDOC_MARKER, X_HEREDOC_MARKER_IGNORE_TABS> {
+    // a line-feed after a heredoc marker indicates a missing start marker tag,
+    // the lexer must not remain in the heredoc marker state
+    {LineTerminator}             { heredocState().removeMarker(zzCurrentPos); backToPreviousState(); return LINE_FEED; }
```

Comment:
I think that there is no need to do `heredocState().removeMarker(zzCurrentPos);` because `heredocState()` is empty otherwise we would not be in this state because heredoc marker move us to state `S_HEREDOC_EXPECTED`
If we reach this line mean there was no marker push.

Reply:
@mmazurkevich 
This PR is mostly to address issues with incremental (highlighting) lexing. The state isn't reset when lexing restarts. This is a safeguard to make sure that we're removing any possibly kept inconsistent state when in incremental lexing mode. (e.g. on a change of `<<EOF` to `<<`).
I'm not sure if such a change could happen with incremental lexing and I'm not able to find a test case for this but would like to keep this as a safeguard if you don't mind.

Unrelated: heredocState() isn't necessarily empty when `removeMarker` is called. There's the possibility of multiple markers on a line, e.g. 
```bash
cat <<EOF <<EOF2
a
EOF
b
EOF2
```
That's not making sense in Bash afaik, but is valid syntax.